ksqldb-parser/src/main/java/io/confluent/ksql/parser/properties/with/CreateSourceProperties.java

```diff
@@ -143,17 +144,25 @@ public String getKafkaTopic() {
   private Map<String, String> getKeyFormatProperties(final String keyFormat, final String name) {
```

Comment:
Just a general question: seems this method is the same as in `CreateSourceAsProperties` ? Wondering if both should share code? Not sure why both are different?

Reply:
Good question. Looks like the they share lots of code and could be refactored to share some base class or common util code. Not sure the reason why they are this way now. Could be good follow up task to refactor?

Reply:
> Could be good follow up task to refactor?

Yeah. Maybe. I guess there might be a reason for the current "split" of both. Maybe @agavra or @vcrfxia would know?

Reply:
they're different object models, and not all the properties are the same - we could probably refactor into a base class (I think I even suggested this once) but Andy was against it at the time and was able to convince me. I don't entirely remember why if I'm being honest 🤷